### ISSUE TITLE
Exit on close and missing icon

### ISSUE THREAD

**Issue by dmaglio** [nylas/N1#465]
I have compiled N1 for Fedora 23, works good but when i click to close N1 exit completly, don't minimize on tray.
Another bug is the missing icon, as you can see in this picture. The tray icon is correct, but not in dash or in application list.
![schermata da 2015-11-25 09-44-00](https://cloud.githubusercontent.com/assets/446573/11392348/6ccd63fc-9359-11e5-9bf6-d3dd051b1a9d.png)

**Comment by dmaglio** [comment]
partial fixed: rpm compilation don't copy resources png

**Comment by jstejada** [comment]
possibly related to #107

**Comment by dmaglio** [comment]
but i have compiled correctly with instruction from #107 and works good. later i try to run with debug info and post result

**Comment by mbilker** [comment]
The instructions from #107 do not account for the icon. Will fix now.

**Comment by mbilker** [comment]
I compile N1 on F23 with Node v0.10.36 and npm v2.14.11 (under NodeSource: Node v5.1.0, npm v3.3.12). Works great after you install `libgit2-devel` and `libgnome-keyring-devel` to satisfy dependencies of native extensions. You also need `rpmdevtools` and `fedora-packager` to build the RPM after building N1.

These instructions are **NOT** for NodeSource repository users. For NodeSource do not install `npm` via `dnf/yum` and do not run the `npm install` commands.

My instructions repeated as follows:
```
# Install dependencies
sudo dnf install rpmdevtools fedora-packager nodejs npm libgit2-devel libgnome-keyring-devel
sudo npm install -g npm@2
sudo npm install -g inherits@2

# Clone repository
git clone https://github.com/nylas/N1

# Build N1 and install to /tmp/nylas-build
script/build (takes a very long time)

# Create RPM from /tmp/nylas-build
rm /tmp/nylas-build/rpm/nylas-*.rpm
script/grunt mkrpm

# Find rpm in /tmp/nylas-build/rpm/
sudo rpm -i /tmp/nylas-build/rpm/nylas-*.rpm
```

**Comment by excieve** [comment]
Doesn't look like #496 helped with this issue.

**Comment by mbilker** [comment]
@excieve If the app is open, then the icon doesn't show. That issue needs to be looked at in code.

**Comment by excieve** [comment]
@mbilker I'd guess `icon` parameter to [BrowserWindow](http://electron.atom.io/docs/v0.27.0/api/browser-window/)'s constructor should fix it. Don't know much about Electron but things like this make the icon display properly with Qt and Gtk.

**Comment by mbilker** [comment]
@excieve Yes. I may try that out. Also, its [BrowserWindow(http://electron.atom.io/docs/v0.35.0/api/browser-window/) because N1 uses Electron v0.35.1.

**Comment by mbilker** [comment]
I may actually need to try this out in a VM since I use i3 as my window manager, which is pretty barebones.

**Comment by daveperdue** [comment]
I submitted a pull request to fix the icon.  The icon path contained an extra `resources` in the path and adding StartupWMClass tells gnome to use the icon in the dock/applications menu.  If you've already got it installed change your .desktop file accordingly.  

*Change:* `Icon=/usr/local/share/nylas/resources/app/nylas.png`
*Add:* `StartupWMClass=Nylas N1`

**Comment by mbilker** [comment]
@daveperdue Alright. Looks valid to me.

Under Arch Linux:
![2015-12-16-16-12-09_593x225](https://cloud.githubusercontent.com/assets/1266541/11854126/cff2c32a-a40f-11e5-857e-94a62a52e8bf.png)

**Comment by jstejada** [comment]
@dmaglio this should be fixed with #726. Reopen if you keep having issues.
